src/jdk.incubator.jextract/share/classes/jdk/incubator/jextract/tool/JavaSourceBuilder.java

```diff
@@ -24,350 +24,175 @@
-                                   .map(SystemABI.Type.class::cast)
-                                   .orElseThrow(()->new AssertionError("Should not get here: " + l));
-            sb.append(TypeTranslator.typeToLayoutName(type));
-        } else if (l instanceof SequenceLayout) {
-            sb.append("MemoryLayout.ofSequence(");
-            if (((SequenceLayout) l).elementCount().isPresent()) {
-                sb.append(((SequenceLayout) l).elementCount().getAsLong() + ", ");
-            }
-            addLayout(((SequenceLayout) l).elementLayout());
-            sb.append(")");
-        } else if (l instanceof GroupLayout) {
-            SystemABI.Type type = l.attribute(NATIVE_TYPE)
-                                   .map(SystemABI.Type.class::cast)
-                                   .orElse(null);
-            if (type == SystemABI.Type.COMPLEX_LONG_DOUBLE) {
-                if (!ABI.equals(SystemABI.ABI_SYSV)) {
-                    throw new RuntimeException("complex long double is supported only for SysV ABI");
-                } else {
-                    sb.append("C_COMPLEX_LONGDOUBLE");
-                }
-            } else {
-                if (((GroupLayout) l).isStruct()) {
-                    sb.append("MemoryLayout.ofStruct(\n");
-                } else {
-                    sb.append("MemoryLayout.ofUnion(\n");
-                }
-                incrAlign();
-                String delim = "";
-                for (MemoryLayout e : ((GroupLayout) l).memberLayouts()) {
-                    sb.append(delim);
-                    indent();
-                    addLayout(e);
-                    delim = ",\n";
-                }
-                sb.append("\n");
-                decrAlign();
-                indent();
-                sb.append(")");
-            }
-        } else {
-            //padding
-            sb.append("MemoryLayout.ofPaddingBits(" + l.bitSize() + ")");
-        }
-        if (l.name().isPresent()) {
-            sb.append(".withName(\"" +  l.name().get() + "\")");
-        }
+    private String displayName(ClassDesc returnType) {
+        return returnType.displayName(); // TODO shorten based on imports
     }
 
-    protected void addVarHandle(String name, Class<?> type, String parentName) {
-        incrAlign();
-        indent();
-        parentName = parentName != null? javaSafeIdentifier(parentName) : parentName;
-        name = javaSafeIdentifier(name);
-        String vhName = parentName != null ?
-                parentName + "$" + name : name;
-        sb.append(PUB_MODS + "VarHandle " + vhName + " = ");
-        if (parentName != null) {
-            addHandlePath(type, parentName, name);
-        } else {
-            addHandlePath(type, name);
-        }
-        sb.append(";\n");
-        decrAlign();
+    private String getFunction(String javaName, FunctionDescriptor fDesc) {
+        return getterCall(constantHelper.addFunctionDesc(javaName, fDesc));
```

Comment:
I've renamed these to `addMethodHandleGetter` and `methodHandleGetCallString` for more clarity, as well as moving the latter to the end of the file to avoid confusion.